### PR TITLE
Add documentation for `snap_getEntropy`

### LINKED ISSUE #584: Document `snap_getEntropy`
Document the `snap_getEntropy` RPC method.

### PR DESCRIPTION
**Pull request by Mrtenz**
Closes #584.

This adds documentation about the `snap_getEntropy` JSON-RPC method.

### CODE REVIEW

File: docs/guide/snaps-rpc-api.md [hunk 1]
```diff
@@ -775,6 +775,70 @@ console.log(dogecoinPublicKey);
...
+
+::: warning Only Callable By
+
+- Snaps
+  :::
+
+#### Parameters
+
+- `Object`
+  - `version` - The literal number `1`. This is to allow for future versions of this method to be added.
```
Comment (legobeat) [inline docs/guide/snaps-rpc-api.md:788]:
```suggestion
  - `version` - The literal number `1`. This is reserved for future use.
```

File: docs/guide/snaps-rpc-api.md [hunk 2]
```diff
@@ -775,6 +775,70 @@ console.log(dogecoinPublicKey);
...
+  - `salt` - An arbitrary string to be used as a salt for the entropy. This can be used to generate different entropy
```
Comment (legobeat) [inline docs/guide/snaps-rpc-api.md:789]: Should we add some more explicit note here about the determinism of the function? While it is described (nicely, I think) in the description, improper salt setting is a common cause of compromise and could warrant a notice for those who average library users who might not grok the implications?

**Review by legobeat — APPROVED** [review]
Left some suggestions to consider

File: docs/guide/snaps-rpc-api.md (hunk 2 above)
Comment (Mrtenz) [reply docs/guide/snaps-rpc-api.md:789]: > improper salt setting and handling are common causes of compromise and could warrant a notice for those who might not grok the implications?

The entropy is Snap-specific. If two Snaps use the same salt, the generated entropy is still different. It shouldn't matter what salt is used.

The implementation is described here: https://github.com/MetaMask/SIPs/blob/60c212d1da607005bef989fd364565f21c66f111/SIPS/sip-6.md

**Comment by FrederikBolding** [comment]
This should not be merged until `0.24.1` is released.

File: docs/guide/snaps-rpc-api.md [hunk 3]
```diff
@@ -775,6 +775,70 @@ console.log(dogecoinPublicKey);
...
+  }
+}
+```
+
+:::
+
+::: tab Code
+
+```javascript
+const entropy = await wallet.request({
```
Comment (FrederikBolding) [inline docs/guide/snaps-rpc-api.md:826]:
```suggestion
const entropy = await snap.request({
```

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**